DM-34187: make auxTel test more robust

suberlak opened the pull request.

jmeyers314 reviewed (commented):
Hi Chris,

This looks okay to me, except that I appear to gain a new collection every time I run `scons` now, which doesn't seem good.  For example, if you run `butler query-collections /repo/main "u/jemeyers/*"` now, you'll see 5 tempfile collections.  It'd be nice if these got deleted when the test passes.

suberlak commented:
I added a `tearDownClass` which does the clean up - when I run it now I don't see any temporary runs in the `/repo/main/$USER` . Rebased against develop,  updated version history against PR128.

jmeyers314 reviewed (approved):
Thanks Chris,
LGTM.
